guides/plugins/plugins/storefront/add-custom-controller.md

```diff
@@ -1,2 +1,215 @@
-# Add custom controller
+# Add custom storefront controller
+
+## Overview
+
+In this guide you'll learn how to create a custom storefront controller.
+
+## Prerequisites
+
+In order to add your own controller for your plugin, you first need a plugin as base.
+Therefore, you can refer to the [Plugin Base Guide](../plugin-base-guide.md).
+
+## Adding custom storefront controller
+
+### Storefront Controller class example
+
+First of all we have to create a new controller which extends from the `StorefrontController` class.
+A controller is also just a service which can be registered via the DI-Container.
+Furthermore, we have to define our `RouteScope` via annotation, it is used to define which domain a route is part of and **needs to be set for every route**.
+In our case the scope is `storefront`.
+
+Go ahead and create a new file `ExampleController.php` in the directory `<plugin root>/src/Storefront/Controller/`.
+
+{% code title="<plugin root>/src/Storefront/Controller/ExampleController.php" %}
+```php
+<?php declare(strict_types=1);
+
+namespace Swag\BasicExample\Storefront\Controller;
+
+use Shopware\Core\Framework\Routing\Annotation\RouteScope;
+use Shopware\Storefront\Controller\StorefrontController;
+
+/**
+ * @RouteScope(scopes={"storefront"})
+ */
+class ExampleController extends StorefrontController
+{
+}
+```
+{% endcode %}
+
+Now we can create a new example method with a `Route` annotation which has to contain our route, in this case it will be `/example`.
+The route defines how our new method will be accessible.
+
+Below you can find an example implementation of a controller method including a route, where we render an `example.html.twig` template file with a template variable `example`.
+
+{% code title="<plugin root>/src/Storefront/Controller/ExampleController.php" %}
+```php
+<?php declare(strict_types=1);
+
+namespace Swag\BasicExample\Storefront\Controller;
+
+use Shopware\Core\Framework\Routing\Annotation\RouteScope;
+use Shopware\Core\System\SalesChannel\SalesChannelContext;
+use Shopware\Storefront\Controller\StorefrontController;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Routing\Annotation\Route;
+
+/**
+ * @RouteScope(scopes={"storefront"})
+ */
+class ExampleController extends StorefrontController
+{
+    /**
+    * @Route("/example", name="frontend.example.example", methods={"GET"})
```

Comment:
the name of the route is important. there are two or three keywords a route name should start with. like frontend. otherwise features like SEO won't work